### ISSUE TITLE
(Time|Duration) is not of kind row

### ISSUE THREAD

**Issue by dangilkerson** [influxdata/flux#2451]
The error "is not of kind row" occurs when two from statements are used with the first containing an aggregateWindow and the second with a range call.

Error: Duration is not of kind row
```
from(bucket: "buck")
  |> aggregateWindow(every: 1m, fn: sum)

from(bucket: "buck")
  |> range(start: -1h)
```

Error: Time is not of kind row
```
from(bucket: "buck")
  |> aggregateWindow(every: 15000ms, fn: last)
  
from(bucket: "buck")
  |> range(start: 2019-10-09T20:44:00Z)
```

**Comment by dangilkerson** [comment]
Could be related to https://github.com/influxdata/flux/issues/2355#issuecomment-581267305

**Comment by nathanielc** [comment]
My thinking on why  #2355 might be related is because this the `start` and `stop` parameters to the range function are type variables, so that they can determine whether the time is a `time` or a `duration`. Since that bug is about generalization and this bug only shows up if you have two independent queries (i.e. two from calls) it seems like we might be mutating the generalized type variable incorrectly under the hood. Just a guess...

**Comment by jlapacik** [comment]
This bug is a result of window being assigned an incorrect type. The fix is here https://github.com/influxdata/flux/pull/2478.

**Comment by jlapacik** [comment]
Fixed by #2478. Closing issue.
